sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/util/DateTimeUtilsSuite.scala

```diff
@@ -508,8 +504,8 @@ class DateTimeUtilsSuite extends SparkFunSuite with Matchers with SQLHelper {
       test("2016-03-13 03:00:00", "PST", "2016-03-13 10:00:00.0")
       test("2016-11-06 00:59:59", "PST", "2016-11-06 07:59:59.0")
       // 2016-11-06 01:00:00 PST could be 2016-11-06 08:00:00 UTC or 2016-11-06 09:00:00 UTC
-      test("2016-11-06 01:00:00", "PST", "2016-11-06 09:00:00.0")
-      test("2016-11-06 01:59:59", "PST", "2016-11-06 09:59:59.0")
+      test("2016-11-06 01:00:00", "PST", "2016-11-06 08:00:00.0")
```

Comment:
See the comment above. `08` or `09` depends on the implementation.

Reply:
This is my only concern.

From what I know, Pacific Standard Time (PST) is UTC-8:00, and Pacific Daylight Time (PDT) is UTC-7:00. So `2016-11-06 09:00:00 UTC` is the only corrected answer.

Reply:
Probably, I have found the reason this behavior. While resolving **PST** to zone offsets, this short name is converted to **America/Los_Angeles** by the map:
https://github.com/frohoff/jdk8u-jdk/blob/master/src/share/classes/java/time/ZoneId.java#L248
And as a consequence, `PST` is not a constant offset:
```scala
   val pst = getZoneId("PST")
    println(pst.getRules.getOffset(LocalDateTime.of(2016, 11, 5, 23, 59, 59)))
    println(pst.getRules.getOffset(LocalDateTime.of(2016, 11, 6, 0, 0, 0)))
    println(pst.getRules.getOffset(LocalDateTime.of(2016, 11, 6, 1, 0, 0)))
    println(pst.getRules.getOffset(LocalDateTime.of(2016, 11, 6, 2, 0, 0)))
    println(pst.getRules.getOffset(LocalDateTime.of(2016, 11, 6, 3, 0, 0)))
```
```
-07:00
-07:00
-07:00
-08:00
-08:00
```
At the test local timestamp `2016-11-06 01:00:00`, the offset is `-07:00`.

Reply:
is it possible to work around this JDK bug in the test? e.g. not use PST

Reply:
@cloud-fan I got `2016-11-06 09:00:00.0` in the test by modifying the map `SHORT_IDS`:
```scala
  def getZoneId(timeZoneId: String): ZoneId = {
    import scala.collection.JavaConverters._

    val m = Map("PST" -> "-08:00")
    ZoneId.of(timeZoneId, m.asJava)
  }
  def getDefaultZoneId(): ZoneId = {
    getZoneId(defaultTimeZone().getID)
  }
```

Reply:
The following works:
```scala
    withDefaultTimeZone(TimeZone.getTimeZone("-08:00")) {
      test("2016-11-06 01:00:00", "-08:00", "2016-11-06 09:00:00.0")
      test("2016-11-06 01:59:59", "-08:00", "2016-11-06 09:59:59.0")
    }
```
@cloud-fan Should I replace `PST` by `-08:00` in the test?

Reply:
yes please

Reply:
done